lib/toggle.js

```diff
@@ -13,9 +13,14 @@ async function toggle (context) {
   const body = await migrate(context)
 
   // 3. Match issue patterns
-  const deps = await match({ body })
+  let deps = await match({ body })
 
-  // 4. Add or remove the label
+  // 4. Remove all dependencies that refer to the same PR.
+  deps = deps.filter(function (number) {
```

Comment:
Excellent! This should work fine. We will need to do the same in `lib/update.js`  though

https://github.com/ahmed-taj/dep/blob/4ec93bcf2429553bc5b960ab212161bd75ad1f6f/lib/update.js#L36-L38

I think it would be better to move the filtering to `lib/helpers/match.js` just before the return statement

https://github.com/ahmed-taj/dep/blob/4ec93bcf2429553bc5b960ab212161bd75ad1f6f/lib/helpers/match.js#L11

That may break some tests. Don't hesitate to ask for help ;)

Reply:
Will do.
Although I think we shouldnt move the filtering code to match.js
IMO match.js serves the purpose of matching all external dependencies.
What say?

Reply:
That's fine. `lib/match`'s purpose is to retrieve and transform dependencies into a consumable form (e.g. Array of numbers right now). I think of filtering as a kind of transformation.

Reply:
Makes sense.
I will make the required fix and do a pull asap :+1:

Reply:
To transfer the filter code to `match.js` I will need a dependency to `payload.number` in match.js.
Is that allright?

Reply:
Yes, you will need to pass the PR's number to `match({body, number})`. for `toggle.js` you can get the number from `issue.number`

https://github.com/ahmed-taj/dep/blob/12342ebba6a5c3278bcb255f4b73535dc5b95898/lib/toggle.js#L8

Same for  `update.js` 

https://github.com/ahmed-taj/dep/blob/12342ebba6a5c3278bcb255f4b73535dc5b95898/lib/update.js#L27